runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -357,9 +356,7 @@ public void init(@Nullable final Applet client) throws Exception
 					saveClientBoundsConfig();
 					runelite.shutdown();
 				},
-				() -> client != null
-					&& client instanceof Client
-					&& ((Client) client).getGameState() != GameState.LOGIN_SCREEN);
+				config.confirmOnClose());
```

Comment:
This should still be lambda, so the boolean is read only when needed (e.g when configuration changes)